res/controllers/Pioneer-DDJ-SB-scripts.js

```diff
@@ -0,0 +1,888 @@
+        PioneerDDJSB.bindSamplerControlConnections('[Sampler' + samplerIndex + ']', isUnbinding);
+    }
+
+    for (fxUnitIndex = 1; fxUnitIndex <= 2; fxUnitIndex++) {
+        PioneerDDJSB.bindFXControlConnections('[EffectRack1_EffectUnit' + fxUnitIndex + ']', isUnbinding);
+    }
+
+    for (channelIndex = 1; channelIndex <= 2; channelIndex++) {
+        PioneerDDJSB.bindDeckControlConnections('[Channel' + channelIndex + ']', isUnbinding);
+    }
+};
+
+PioneerDDJSB.setDeckSoftTakeover = function (channel, isUnbinding) {
+    engine.softTakeover(channel, "volume", !isUnbinding);
+    engine.softTakeover(channel, "rate", !isUnbinding);
+    engine.softTakeover(channel, "pregain", !isUnbinding);
+    engine.softTakeover(channel, "filterHigh", !isUnbinding);
+    engine.softTakeover(channel, "filterMid", !isUnbinding);
+    engine.softTakeover(channel, "filterLow", !isUnbinding);
+    engine.softTakeover("[QuickEffectRack1_" + channel + "]", "super1", !isUnbinding);
+};
+
+PioneerDDJSB.setAllSoftTakeover = function (isUnbinding) {
+    var channelIndex;
+    for (channelIndex = 1; channelIndex <= 4; channelIndex++) {
+        PioneerDDJSB.setDeckSoftTakeover('[Channel' + channelIndex + ']', isUnbinding);
+    }
+};
+
+
+///////////////////////////////////////////////////////////////
+//                       DECK SWITCHING                      //
+///////////////////////////////////////////////////////////////
+
+PioneerDDJSB.deckSwitchTable = {
+    '[Channel1]': '[Channel1]',
+    '[Channel2]': '[Channel2]'
+};
+
+PioneerDDJSB.channelRegEx = /\[Channel(\d+)\]/;
+
+PioneerDDJSB.deckToggleButton = function (channel, control, value, status, group) {
+    var deckNumber = parseInt(PioneerDDJSB.channelRegEx.exec(PioneerDDJSB.deckSwitchTable[group])[1], 10);
+
+    if (value) {
+        if (deckNumber <= 2) {
+            deckNumber += 2;
+        } else {
+            deckNumber -= 2;
+        }
+        PioneerDDJSB.deckSwitchTable[group] = '[Channel' + deckNumber + ']';
+        PioneerDDJSB.initDeck(PioneerDDJSB.deckSwitchTable[group], false);
+        PioneerDDJSB.setDeckSoftTakeover(PioneerDDJSB.deckSwitchTable[group], true);
+        PioneerDDJSB.setDeckSoftTakeover(PioneerDDJSB.deckSwitchTable[group]);
+    }
+};
+
+PioneerDDJSB.initDeck = function (group, isFirstBinding) {
+    var disconnectDeck = parseInt(PioneerDDJSB.channelRegEx.exec(group)[1]);
+
+    if (disconnectDeck <= 2) {
+        disconnectDeck += 2;
+    } else {
+        disconnectDeck -= 2;
+    }
+
+    if (!isFirstBinding) {
+        PioneerDDJSB.bindDeckControlConnections('[Channel' + disconnectDeck + ']', true);
+    }
+
+    PioneerDDJSB.bindDeckControlConnections(group, false);
+    PioneerDDJSB.nonPadLedControl(group, PioneerDDJSB.nonPadLeds.shiftKeyLock, disconnectDeck <= 2);
+    PioneerDDJSB.triggerVinylLed(PioneerDDJSB.channelGroups[group]);
+};
+
+PioneerDDJSB.toggleEngineValue = function (group, control) {
```

Comment:
IMHO implementing this in common-controller-scripts.js is a good thing but needs to be in a separate PR.